Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -203,6 +203,154 @@ namespace impl
+        const web::json::field_as_string label_property{ U("label") };
+        const web::json::field_as_string manifest_href_property{ U("manifest_href") };
+        const web::json::field_as_array  subscription_property{ U("subscription") };
+        const web::json::field_as_value  tags_property{ U("tags") };
+        const web::json::field_as_string transport_property{ U("transport") };
+        const web::json::field<nmos::tai> version_property{ U("version") };
+
+        // Define property descriptors: required fields
+        std::vector<web::json::value> snd_control_property_descriptors = {
+            nmos::experimental::make_control_class_property_descriptor(U("description"), { 3, 1 }, description_property, U("NcString")),
+            nmos::experimental::make_control_class_property_descriptor(U("resource_id"), { 3, 2 }, uuid_property, U("NcUuid")),
+            nmos::experimental::make_control_class_property_descriptor(U("label"), { 3, 3 }, label_property, U("NcString")),
+            nmos::experimental::make_control_class_property_descriptor(U("tags"), { 3, 4 }, tags_property, U("NcSenderTagsType")),
+            nmos::experimental::make_control_class_property_descriptor(U("version"), { 3, 5 }, version_property, U("NcVersionCode")),
+        };
+        // Property descriptor for optional fields
+        if (sender_data.has_field(U("caps"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("caps"),{ 3, 6 }, caps_property, U("NcSenderCapsType")));
+        if (sender_data.has_field(U("device_id"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("description"), { 3, 7 }, device_id_property, U("NcUuid")));
+        if (sender_data.has_field(U("flow_id"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("flow_id"), { 3, 8 }, flow_id_property, U("NcUuid")));
+        if (sender_data.has_field(U("interface_bindings"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("interface_bindings"), { 3, 9 }, interface_bindings_property, U("NcArray")));
+        if (sender_data.has_field(U("manifest_href"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("manifest_href"), { 3, 10 }, manifest_href_property, U("NcUri")));
+        if (sender_data.has_field(U("subscription"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("subscription"), { 3, 11 }, subscription_property, U("NcSenderObjectType")));
+        if (sender_data.has_field(U("transport"))) snd_control_property_descriptors.push_back(nmos::experimental::make_control_class_property_descriptor(U("transport"), { 3, 12 }, transport_property, U("NcString")));
+
+        auto snd_class_id = nmos::nc::make_class_id(nmos::nc_worker_class_id, 0, { 5 });
```

Comment:
`make_audio_sender_descriptors` actually performs two functions - as well as making the audio sender class and datatype descriptors, it also makes the class_id for the audio sender class - I think I'd rather see the class id passed in as a parameter and then the function is only doing one thing.